supplier_stock/models/supplier_stock.py

```diff
@@ -36,6 +36,11 @@ class SupplierStock(models.Model):
         string='Product',
         required=True,
     )
+    product_name = fields.Char(
+        string='Product Name',
+        related='product_id.product_tmpl_id.name',
+        store=True,
```

Comment:
I guess you want to add `readonly=True`?  You can change the product name through supplier stock screen, otherwise, unless that is what you want.